### PR TITLE
Feature Task #1 UI

### PR DESCRIPTION
**Pull request by nikkaktus24**

### CODE REVIEW

File: src/app/containers/Tasks/index.tsx [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+            return assignments.map((item: any, i: number) => {
+                const props = {
+                    title: item.title,
+                    urlToDescription: item.urlToDescription,
+                    taskId: item.taskId,
+                    studentId: item.studentId,
+                    status: item.status,
+                    score: item.score,
+                    courseId: item.courseId,
+                    submit: submitTask,
```
Comment (yuliaHope) [inline src/app/containers/Tasks/index.tsx]: You can use destructuring here to avoid unused properties: `const { mentorId, ...assignment} = item` and pass this `assignment` to component.

File: src/app/components/taskForm/FooterForm.tsx [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+                            label="Comments"
+                            placeholder="Write comments here"
+                            component={ReduxFormInput}
+                            required={false}
+                            type="textarea"
+                            className="form-control form-control-sm"
+                        />
+                    </FormGroup>
+                    <Button type="submit" color="primary" size="sm">
+                        Submit
```
Comment (yuliaHope) [inline src/app/components/taskForm/FooterForm.tsx]: Please don't use `type="submit"` for button. Use `type="button"` instead.

File: src/app/components/taskForm/FooterForm.tsx [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+
+    handleChange = (event: any) => {
+        this.setState({ [event.target.name]: event.target.value });
+    };
+
+    render() {
+        const { handleSubmit } = this.props;
+        return (
+            <small className="text-muted">
+                <Form onSubmit={handleSubmit(this.handleFormSubmit)}>
```
Comment (yuliaHope) [inline src/app/components/taskForm/FooterForm.tsx]: this is looks so strange for me... why do you pass function to function here? And I can't find any place where you pass this prop to this component.

File: src/app/containers/Tasks/index.tsx [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+
+const mapDispatchToProps = (dispatch: any, props: any): AssigmentContainerProps => {
+    return {
+        ...props,
+        onLoad: (courseId: string) => {
+            dispatch(fetchAssignments(courseId));
+        },
+        submitTask: (assignment: IAssignment) => {
+            dispatch(submitSolution(assignment));
+        },
```
Comment (yuliaHope) [inline src/app/containers/Tasks/index.tsx]: Please move `mapDispatchToProps` and `mapStateToProps` between component and connect.

File: src/app/containers/Tasks/index.tsx [hunk 5]
```diff
@@ -0,0 +1,124 @@
...
+};
+
+type AssigmentContainerProps = {
+    onLoad: (courseId: string) => void;
+    submitTask: (assignment: IAssignment) => void;
+    studentId: string;
+    courseId: string;
+    isLoading: boolean;
+    error: boolean | undefined;
+    assignments: any;
```
Comment (yuliaHope) [inline src/app/containers/Tasks/index.tsx]: Why `any`? Please specify type for assignments.

File: src/app/containers/Tasks/index.tsx [hunk 6]
```diff
@@ -0,0 +1,124 @@
...
+
+const Temp = {
+    github: 'fdsfsd',
+    score: 100,
+};
+
+class Tasks extends React.Component<AssigmentContainerProps> {
+    constructor(props: AssigmentContainerProps) {
+        super(props);
+    }
```
Comment (yuliaHope) [inline src/app/containers/Tasks/index.tsx]: Why did you add constructor here?

File: src/app/containers/Tasks/index.tsx [hunk 7]
```diff
@@ -0,0 +1,124 @@
...
+                                <p>You are in the TOP 50 students!</p>
+                                <p>Full Score: {Temp.score}</p>
+                            </div>
+                        </div>
+                        <CardDeck>
+                            {(() => {
+                                if (!isLoading && !error) {
+                                    return this.generateTasks();
+                                }
+                            })()}
```
Comment (yuliaHope) [inline src/app/containers/Tasks/index.tsx]: Why IIFE here???

File: src/app/core/actions/assignment.ts [hunk 8]
```diff
@@ -0,0 +1,45 @@
...
+import { IAssignment } from '../models';
+
+export function fetchAssignments(courseId: string) {
+    return async (dispatch: any) => {
+        dispatch({
+            type: ASSIGNMENT.LOADING,
+        });
+
+        try {
+            const assignmentsResult = await getAssignmentsById(courseId);
```
Comment (yuliaHope) [inline src/app/core/actions/assignment.ts]: Might it would be better to name this function `getAssignmentsByCourseId`?

File: src/app/core/actions/__tests__/assignment.spec.tsx [hunk 9]
```diff
@@ -0,0 +1,49 @@
...
+                        completeDate: 0,
+                        checkDate: 0,
+                        status: AssignmentStatus.Assigned,
+                    }),
+                )
+                .then(() => expect(store.getActions()).toContainEqual({ type: ASSIGNMENT.LOADING }))
+                .then(() => expect(store.getActions()).toContainEqual({ type: ASSIGNMENT.SUBMIT_SOLUTION_OK }));
+        });
+    });
+});
```
Comment (yuliaHope) [inline src/app/core/actions/__tests__/assignment.spec.tsx]: Why didn't you test `FAIL` cases? To test you actions in proper way you should do mocks for api calls. See jest.mock() documentation for details https://jestjs.io/docs/en/mock-functions#using-a-mock-function

File: src/app/core/actions/assignment.ts [hunk 10]
```diff
@@ -0,0 +1,45 @@
...
+}
+
+export function submitSolution(assignment: IAssignment) {
+    return async (dispatch: any) => {
+        dispatch({
+            type: ASSIGNMENT.LOADING,
+        });
+
+        try {
+            const assignmentResult = await submitSolutionApi(assignment);
```
Comment (yuliaHope) [inline src/app/core/actions/assignment.ts]: just `updateAssignment` for example and without `Api` postfix. Your function name should say everybody what this function do

Comment (yuliaHope) [reply src/app/core/actions/assignment.ts]: the same for variable names

File: src/app/core/api/schedule.ts [hunk 11]
```diff
@@ -49,14 +49,14 @@ export function deleteStageApi(id: string) {
...
 
-export function addEventApi(event: IEvent) {
-    return axios.post<EventPostResponse>(`/api/event`, event).then(response => response.data);
+export function addEventApi(event: IEvent, route: string) {
+    return axios.post<EventPostResponse>(`/api/${route}`, event).then(response => response.data);
 }
 
-export function updateEventApi(event: IEventDocument) {
-    return axios.patch<EventPatchResponse>(`/api/event`, event).then(response => response.data.data);
+export function updateEventApi(event: IEventDocument, route: string) {
```
Comment (yuliaHope) [inline src/app/core/api/schedule.ts]: Please rename `route` to smth more understandable and more readable.

File: src/app/components/taskForm/FooterForm.tsx (hunk 3 above)
Comment (nikkaktus24) [reply src/app/components/taskForm/FooterForm.tsx]: It's one of the conditions of using Form from redux-form.
https://stackoverflow.com/questions/43966804/how-to-handlesubmit-with-a-redux-form

File: src/app/components/taskForm/__tests__/FooterForm.spec.tsx [hunk 12]
```diff
@@ -0,0 +1,10 @@
+import { shallow } from 'enzyme';
+import * as React from 'react';
+import FooterForm from '../FooterForm';
+
+describe('FooterForm', () => {
+    it('renders correctly if no data', () => {
+        const output = shallow(<FooterForm taskId={0} courseId={''} submitApi={jest.fn()} />);
+        expect(output).toMatchSnapshot();
+    });
```
[the remark this answers is not in the document]
Comment (nikkaktus24) [reply src/app/components/taskForm/__tests__/FooterForm.spec.tsx]: I tried to do it. I always get this error, it's linked with redux-form input. I tried to search something in google, but it wasn't bring result.
image

File: src/app/containers/Tasks/index.tsx [hunk 13]
```diff
@@ -0,0 +1,124 @@
...
+const cn = classNames(require('./index.scss'));
+
+const mapStateToProps = (state: RootState, props: any): AssigmentContainerProps => {
+    if (state.assignments == null) {
+        return props;
+    }
+    return {
+        ...props,
+        isLoading: state.assignments.isLoading,
+        courseId: props.match.params.id,
```
[the remark this answers is not in the document]
Comment (yuliaHope) [reply src/app/containers/Tasks/index.tsx]: Props throws to component without connect, (container is component in any case) and it works perfectly without that. Please, remove it.
